tufrepo/git_repo.py

```diff
@@ -39,28 +39,36 @@
 }
 
 
-class GitRepository(Repository):
-    """Manages loading, saving (signing) repository metadata in files stored in git"""
+def _git(command: List[str]):
+    """Helper to run git commands in the repository git repo"""
+    full_cmd = ["git"] + command
+    proc = subprocess.run(full_cmd)
+    return proc.returncode
 
-    def __init__(self, keyring: Keyring):
+class GitMetadata(MetadataDescriptor):
+    def __init__(self, role: str, keyring: Keyring, init: bool = False):
+        self._role = role
         self._keyring = keyring
+        self._fname = self._get_filename(role)
+        if init:
+            # safety check
+            filenames = glob.glob(f"*.{role}.json")
+            versions = [int(name.split(".", 1)[0]) for name in filenames]
+            if versions:
+                raise ValueError(f"cannot initialize {role} as versions already exist")
```

Comment:
I think there are some duplications between this portion of `GitMetadata.__init__` and `GitMetadata._get_filename`: https://github.com/vmware-labs/repository-editor-for-tuf/blob/a753029c1b0cb5631d8de9a8e422522a6e52f24f/tufrepo/git_repo.py#L74
Do you think I am right or do we need both checks?

Reply:
both are needed (not that the code in `__init__()` only executes when `init==True`) so you can't just move that to `_get_filename()`. They do use the same two-liner to find currently stored versions of the role.